src/main/java/seedu/pennywise/model/entry/Date.java

```diff
@@ -6,19 +6,22 @@
 import java.time.LocalDate;
 import java.time.YearMonth;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
 
 /**
  * Represents an {@code Entry}'s date in the PennyWise application.
  * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
  */
 public class Date {
     public static final String MESSAGE_CONSTRAINTS =
-            "Dates should be of the format dd-mm-yyyy and it should only contain numbers";
-    public static final String VALIDATION_PATTERN = "dd-MM-yyyy";
+            "Dates should be of the format dd-mm-yyyy and it should only contain numbers,\n "
+                    + "Date must also be a valid date";
+    public static final String VALIDATION_PATTERN = "dd-MM-uuuu";
     public static final String VALIDATION_REGEX =
             "^([0-2][0-9]||3[0-1])-(0[1-9]||[1-9]||1[0-2])-([0-9][0-9])?[0-9][0-9]$";
 
-    public static final String YEAR_MONTH_PATTERN = "yyyy-MM";
+    public static final String YEAR_MONTH_PATTERN = "uuuu-MM";
```

Comment:
Thanks for fixing the bug! https://stackoverflow.com/questions/41177442/uuuu-versus-yyyy-in-datetimeformatter-formatting-pattern-codes-in-java